### ISSUE TITLE
Feature request - @tagging other items in a note/the pad, diagram arrows withing a note pointing to the tagged item.

### ISSUE THREAD

**Issue by petteremil** [MicroPad/MicroPad-Core#462]
**Is your feature request related to a problem? Please describe.**
Organizing elements inside a note, to show links and such.
The usecase I'm thinking of is documenting code, so a simple example would be if you have some functions/classes that interacts with other classes/functions, then it would be nice to display this relation somehow.. 

**Describe the solution you'd like**
maybe an arrow between text/elements in a note. Not sure what the best implementation would be in this structure that exists..
Maybe there could be a direct link in the text, you have hashtags now, but say you could link to a text/item directly with @ or something, then an arrow showing the direction (so if you tag @y in x, the arrow would point towards @y from x, and if you had an @x in y at the same time, the arrow would be bi-directional..). I guess that if this was to be possible, each note would need to have some sort of unique name/id you could reference, I assume it could be unique only within a note, if you limit the tagging to within the note. otherwise if you want tagging to be allowed throughout the pad, then maybe you add a prefix for the note or section+note too, of course the arrow system would only work inside a note, but clicking on the tag to go to another note and item would probably be nice to be able to do.

**Describe alternatives you've considered**
I guess you could just use a drawing to denote information flow between two items, but that gets complicated as soon as you move one of the items in the note, so something that just kept pointing correctly would be great, like a typical diagram arrow system.

**Additional context**
If someone picks this up, they could make it more interesting to code, if they add functionality for the arrows to go around other objects instead of just taking the straight path across anything else to the tagged object. Also, if you implement tagging globally, then another interesting challenge could be to jump to the coordinates of the item you clicked the link of.

Hope this hasn't been suggested already, I tried looking at the boards and issues, and couldn't find anything except the newly added hashtags, so I thought I'd at least suggest it 😊

**Comment by NickGeek** [comment]
Thanks for the suggestion! When you say "note" are you referring to one markdown element or the whole collection of elements? In either case, both have unique identifiers in the structure of a notebook (notebooks are stored as a flat data structure internally, basically a bunch of adjacency lists between sections/notes/elements).

Anyway, what you're suggesting would probably force us to allow users to edit those values so they can `@tag` them. A potentially easier solution might be to just make it all work like a hetero-`#tag`:

**Element 1 contains:**
`!@Analysis`

**Elements 2 and 3 contain:**
```
@Analysis
```

Unlike hashtags which are bi/multidirectional (if only 2 hashtags exist they work as a link, otherwise it prompts a search) and global across notebooks. The `@` tag references a declaration (`!@`) within the same note. If multiple `!@` exist, probably show a toast explaining that it can't make the jump due to ambiguity.

Would that support the functionality you want?

> except the newly added hashtags
I think hashtags have been in MicroPad since 2017 or 2018 ;)

**Comment by petteremil** [comment]
Thank you for replying! 😊
Also, while I'm at it! really like the app so far, it ticks all the boxes for me otherwise, and I'll probably be using it for a long time going forward 💪 Once I have some proper time off and understand the app a bit more, maybe I could contribute some to the repo down the line 🤞

By "note" I mean the whole collection containing the Text, Images, Drawings and so on.

Aah, I like your Idea a lot! it keeps it even more simple than having to expose IDs and so on! 😊
I guess it would be nice if you could get a suggestion box or something of that kind when starting to type "@" maybe after the first following character, and if the @ wasn't prepended by any text. that should make it not mess with users writing up emails, and not show all possible matches on typing just "@".

I would love it if it was possible to get some kind of visual pointer from one "element/object" in the note to another (maybe you could toggle it on/off if some people feel like it makes it too cluttered 🤷‍♂️), as it would make it easy to see relations between different things, but I also understand that it would increase the scope of adding such a thing a lot bigger, and possibly go against how you intend the feel of the notes should be. So I'll gladly respect whatever you think is correct in terms of the vision of the app.

Just to illustrate what I mean, so it's not unclear:

![image](https://github.com/user-attachments/assets/dafcf8b7-9dfc-4b03-bca8-de1c62be2ec2)


But anyhows, the hetero-tagging would be super nifty no matter what! 😊

> I think hashtags have been in MicroPad since 2016 or 2017 ;)

Ahaa, I was fooled by the top "done" card in the "MicroPad's DevBad so PlanHad" board 🙈 saw it at the very top, and thought it was recent 😅

**Comment by petteremil** [comment]
Just to throw it in before I close excalidraw, this would probably be a cleaner implementation if there is a tag both ways 😅

![image](https://github.com/user-attachments/assets/034b3260-85f4-4e68-ab77-f39de26c865d)

**Comment by NickGeek** [comment]
Makes sense, thanks for the drawings. I'm currently in the final stages of writing my thesis so I probably won't be able to do anything on this until February-- but if you have any questions feel free to let me know :)

**Comment by petteremil** [comment]
Sorry for the late reply! 
Just wanted to say good luck on your thesis! 💪💪😊

**Comment by NickGeek** [comment]
Thanks :)
